smack-websocket-java11/src/main/java/org/jivesoftware/smack/websocket/java11/Java11WebSocket.java

```diff
@@ -139,11 +139,13 @@ protected void send(String element) {
 
     @Override
     public void disconnect(int code, String message) {
-        CompletableFuture<WebSocket> completableFuture = webSocket.sendClose(code, message);
         try {
-            completableFuture.get();
+            if (!webSocket.isOutputClosed()) {
+                CompletableFuture<WebSocket> completableFuture = webSocket.sendClose(code, message);
+                completableFuture.get();
+            }
         } catch (ExecutionException e) {
-            onWebSocketFailure(e);
+            LOGGER.log(Level.WARNING, "Failed to send final close when disconnecting " + this, e);
```

Comment:
ErrorProne does not like the 'this' reference in the message string:
> warning: [ObjectToString] Java11WebSocket is final and does not override Object.toString, so converting it to a string will print its identity (e.g. `Java11WebSocket@4488aabb`) instead of useful information.